src/ui/components/DeckBuilder/DeckBuilder.tsx

```diff
@@ -75,9 +80,9 @@ export const DeckBuilder = ({ onDone }: DeckBuilderProps) => {
           variant="contained"
           size="large"
           onClick={handleDone}
-          disabled={!isDeckValid}
+          disabled={!isDeckValid || isLoading}
         >
-          Done
+          {isLoading ? <CircularProgress size={24} color="inherit" /> : 'Done'}
```

Comment:
@jules add test coverage to validate that the loading UI presents as expected. Also, add a Storybook story to show the loading state.